BigQuery/tests/Unit/BigQueryClientTest.php

```diff
@@ -415,6 +426,92 @@ public function testCreatesDatasetWithDefaultLocation()
         $this->assertInstanceOf(Dataset::class, $dataset);
     }
 
+    /**
+     * @dataProvider jobConfigDataProvider
+     */
+    public function testRunJob($expectedData, $expectedMethod, $returnedData)
+    {
+        $client = $this->getClient(['location' => self::LOCATION]);
+
+        $jobConfig = $this->prophesize(JobConfigurationInterface::class);
+        $jobConfig->toArray()
+            ->willReturn($expectedData);
+        $this->connection->$expectedMethod($expectedData)
+            ->willReturn($returnedData)
+            ->shouldBeCalledTimes(1);
+
+        $this->connection->getJob(Argument::allOf(
+            Argument::withEntry('projectId', self::PROJECT_ID),
+            Argument::withEntry('jobId', self::JOB_ID)
+        ))
+            ->willReturn([
+                'status' => [
+                    'state' => 'DONE'
+                ]
+            ])
+            ->shouldBeCalledTimes(1);
+
+        $client->___setProperty('connection', $this->connection->reveal());
+        $job = $client->runJob($jobConfig->reveal());
+
+        $this->assertInstanceOf(Job::class, $job);
+        $this->assertTrue($job->isComplete());
+    }
+
+    /**
+     * @dataProvider jobConfigDataProvider
+     */
+    public function testStartJob($expectedData, $expectedMethod, $returnedData)
+    {
+        $client = $this->getClient(['location' => self::LOCATION]);
+
+        $jobConfig = $this->prophesize(JobConfigurationInterface::class);
+        $jobConfig->toArray()
+            ->willReturn($expectedData);
+        $this->connection->$expectedMethod($expectedData)
+            ->willReturn($returnedData)
+            ->shouldBeCalledTimes(1);
+
+        $this->connection->getJob(Argument::any())
+            ->shouldNotBeCalled();
+
+        $client->___setProperty('connection', $this->connection->reveal());
+        $job = $client->startJob($jobConfig->reveal());
+
+        $this->assertInstanceOf(Job::class, $job);
+        $this->assertFalse($job->isComplete());
+        $this->assertEquals($this->jobResponse, $job->info());
+    }
+
+    public function jobConfigDataProvider()
+    {
+        $expected = [
+            'projectId' => self::PROJECT_ID,
+            'jobReference' => [
+                'projectId' => self::PROJECT_ID,
```

Comment:
JobReference can contain a location reference, but it's unclear to me if its needed here.